### PR TITLE
[kmac] Fix usage of sideload key if masking is disabled

### PR DESCRIPTION
**Pull request by vogelpi**
Previously, only Share 0 for the sideload key (always provided in masked form) was used despite the KMAC documentation stating: "KeyMgr provides the sideloaded key in two-share masked form regardless of the compile-time parameter EnMasking. If EnMasking is not defined, the KMAC merges the shared key to the unmasked form before uses the key."

This commit fixes the RTL to XOR the two shares for the unmasked design and align the RTL with the documentation. The masked design is not affected.

This is related to lowRISC/OpenTitan#16550 and reverts commit ded4e7a4200a81ec18f9636f10b5542f1490af23 .

Signed-off-by: Pirmin Vogel <vogelpi@lowrisc.org>

### CODE REVIEW

**Comment by vogelpi** [comment]
Also for the masked implementation, the sideload tests keep passing.

**Comment by cindychip** [comment]
That lgtm, I will close #16550 once it is merged. @eunchan it would be great if you can confirm it :)

**Review by eunchan — APPROVED** [review]

**Comment by vogelpi** [comment]
Thanks for your feedback @cindychip and @eunchan .

**Pull request merged**
